src/context/volume-context.js

```diff
@@ -0,0 +1,82 @@
+import {createContext, useContext, useReducer} from "react";
+
+const VolumeContext = createContext();
+
+// Custom component to provide the Volume context
+function VolumeProvider(props) {
+  const [state, dispatch] = useReducer(volumeReducer, {
+    toggles: {
```

Comment:
`toggles` to me implies an on/off value and while right now there are two of each of these things, that is not necessarily the case moving forward.  I would make this hold the actual value (e.g. `measurement: 'temp'`) instead of a mapped integer and name it something more like maybe `dataset` or `options` or similar.

Reply:
I renamed it `options`